neuro_flow/context.py

```diff
@@ -338,6 +338,20 @@ async def client(self) -> AsyncIterator[Client]:
         yield self._client
 
 
+_MODULE_PARENT = TypeVar("_MODULE_PARENT", bound=RootABC, covariant=True)
```

Comment:
I've implemented the `ModuleContext` as a generic class mostly for early expression validation.